lightly/models/utils.py

```diff
@@ -498,7 +498,7 @@ def nearest_neighbors(
         They both have size: [batch_size, input_map_size, feature_dimension]
     """
 
-    if num_matches is None or num_matches == -1:
+    if num_matches is None or num_matches == -1 or num_matches > input_maps.size(1):
```

Comment:
Would it make sense to raise an error if `num_matches > input_maps.size(1)`?

Reply:
I thought about that, but to be honest, if something like that happens I would love not to get interrupted. Like the -1. If I put more matches, it makes sense to just take them all imho.

Reply:
The `num_matches` parameter is useful only if it is less than `input_maps.size(1)`. Otherwise, I would just not interrupt stuff.